### PR TITLE
chore(druid): add persistence option for indexers

### PR DESCRIPTION
**Pull request by spnngl**

### CODE REVIEW

File: charts/druid/templates/indexer/statefulset.yaml [hunk 1]
```diff
@@ -131,10 +131,44 @@ spec:
...
       volumes:
         - name: data
           emptyDir:
-            sizeLimit: "{{ $categoryValue.emptyDir.size }}"
+            {{- with $categoryValue.emptyDir }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+      {{- end }}
+  {{- if $categoryValue.persistence.enabled }}
+  # XXX alpha support in 1.23, beta & enabled by default in 1.27
```
Comment (desaintmartin) [inline charts/druid/templates/indexer/statefulset.yaml:143]: Probably not xxx

File: charts/druid/templates/indexer/statefulset.yaml [hunk 2]
```diff
@@ -131,10 +131,44 @@ spec:
...
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  volumeClaimTemplates:
+    - kind: PersistentVolumeClaim
+      apiVersion: v1
+      metadata:
+        name: data
+        namespace: {{ .Release.Namespace }}
+        labels:
+          app: indexer
```
Comment (desaintmartin) [inline charts/druid/templates/indexer/statefulset.yaml:159]: {{- include "druid.common-labels" $ | nindent 4 }} and co ?

File: charts/druid/templates/indexer/statefulset.yaml (hunk 1 above)
Comment (spnngl) [reply charts/druid/templates/indexer/statefulset.yaml:143]: removed

File: charts/druid/templates/indexer/statefulset.yaml (hunk 2 above)
Comment (spnngl) [reply charts/druid/templates/indexer/statefulset.yaml:159]: done

**Pull request merged**